app/src/main/java/com/ustwo/boilerplate/base/BaseActivity.java

```diff
@@ -0,0 +1,32 @@
+package com.ustwo.boilerplate.base;
+
+import android.os.Bundle;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+public abstract class BaseActivity<V extends BaseView> extends AppCompatActivity {
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(getLayoutId());
+    getPresenter().onViewAttached(getPresenterView());
+  }
+
+  @Override
+  protected void onDestroy() {
+    getPresenter().onViewDetached();
```

Comment:
Data is typically cached at some level so not sure helps to have data subscriptions lingering. Nissan works this way. 

It's often helpful to have events happen when the view is no longer visible. Say Activity A starts some intensive animation on its Views or plays a video then we press HOME button. We'd want that animation or video to stop when the view is paused. To achieve this with the current design we'd need more callbacks.

Additionally when you have a presenter for a custom view (no Activities) you get `onWindowVisibilityChanged` events that fire in `onPause` and `onResume`. So it's possible to achieve the same presenter lifecycle calls regardless of what is providing the View (and no pass down from Activity to leaf Views required either).

I try